### PR TITLE
2.0 implementation work

### PR DESCRIPTION
**Pull request by jmartisk**
This includes most 2.0 work and some extra fixes and refactoring
- Multi-dimensional tags implementation, that's the biggest part
- Adds `UnspecifiedMetadata` class to mark cases where a new metric is being requested but the user didn't specify metadata, so it should copy metadata from an already existing metric with the same name
- Changes in exporters related to the new tags
- Tags for base metrics `gc.count` and `gc.time`
- Fixes for #67 #68 #42 #43 #38 
- Renaming all Prometheus references to OpenMetrics
- And a lot of new tests

### CODE REVIEW

**Comment by pilhuhn** [comment]
retest please

**Comment by jmartisk** [comment]
retest this please

File: implementation/src/main/java/io/smallrye/metrics/MetricsRegistryImpl.java [hunk 1]
```diff
@@ -57,31 +54,28 @@
 @Vetoed
 public class MetricsRegistryImpl extends MetricRegistry {
 
-    private static final Logger log = Logger.getLogger(MetricsRegistryImpl.class);
-    private static final String NONE = "NONE";
+    private static Logger log = Logger.getLogger(MetricsRegistryImpl.class);
```
Comment (pilhuhn) [inline implementation/src/main/java/io/smallrye/metrics/MetricsRegistryImpl.java]: Why did you remove the `final`?

Comment (jmartisk) [reply implementation/src/main/java/io/smallrye/metrics/MetricsRegistryImpl.java]: I have no idea :)
